tests/python/test_dataset_conversion.py

```diff
@@ -0,0 +1,610 @@
+    cdataname = ["x"]
+    cdimname = ["dim_data_x"]
+    cattrname = ["x_coord_units","x_coord_longname"] 
+   
+    # Calling method add_metadata_for_xarray on 1D dataset 
+    # good data and prerequisite for transform_to_xarray 
+    with pytest.raises(RedisKeyError): 
+        DatasetConverter.add_metadata_for_xarray(
+            ds1, 
+            data_names=["baddata"],
+            dim_names=dimname,
+            coord_names=coordname,
+            attr_names=attrname
+        )
+        # Calling method add_metadata_for_xarray for x coordinate
+        DatasetConverter.add_metadata_for_xarray(
+            ds1,
+            data_names=cdataname, 
+            dim_names=cdimname,
+            attr_names=cattrname 
+        )
+        d1_xarray_ret = DatasetConverter.transform_to_xarray(ds1)
+
+    with pytest.raises(RedisKeyError): 
+        DatasetConverter.add_metadata_for_xarray(
+            ds1, 
+            data_names=dataname,
+            dim_names=["baddata1","baddata2"],
+            coord_names=coordname,
+            attr_names=attrname
+        )
+        # Calling method add_metadata_for_xarray for x coordinate
+        DatasetConverter.add_metadata_for_xarray(
+            ds1,
+            data_names=cdataname, 
+            dim_names=cdimname,
+            attr_names=cattrname 
+        )
+        d1_xarray_ret = DatasetConverter.transform_to_xarray(ds1)
+    
+    with pytest.raises(RedisKeyError): 
+        DatasetConverter.add_metadata_for_xarray(
+            ds1, 
+            data_names=dataname,
+            dim_names=dimname,
+            coord_names=["baddata"],
+            attr_names=attrname
+        )
+        # Calling method add_metadata_for_xarray for x coordinate
+        DatasetConverter.add_metadata_for_xarray(
+            ds1,
+            data_names=cdataname, 
+            dim_names=cdimname,
+            attr_names=cattrname 
+        )
+        d1_xarray_ret = DatasetConverter.transform_to_xarray(ds1)
+        
+    with pytest.raises(RedisKeyError): 
+        DatasetConverter.add_metadata_for_xarray(
+            ds1, 
+            data_names=dataname,
+            dim_names=dimname,
+            coord_names=coordname,
+            attr_names=["baddata1","baddata2","baddata3"]
+        )
+        # Calling method add_metadata_for_xarray for x coordinate
+        DatasetConverter.add_metadata_for_xarray(
+            ds1,
+            data_names=cdataname, 
+            dim_names=cdimname,
+            attr_names=cattrname 
+        )
+        d1_xarray_ret = DatasetConverter.transform_to_xarray(ds1)
+        
+#------- beginning of 2d transform_to_xarray tests-------
+
+def test_transform_to_xarray_2d():
+    """Test transform_to_xarray method with correct 2d
+    tensor data and assert equality
+    """
+    
+    # Create 2d Dataset 
+    ds2 = Dataset("ds-2d")
+    # add data to Datset
+    add_2d_data(ds2)
+    # Calling add_metadata_for_xarray for 2D data
```

Comment:
Globally, please convert from progressive tense (calling) to imperative tense (call).  This is the way most people write comments, so seeing it done a different way is distracting. (There's nothing inherently wrong with what you've done, it's just not the standard.)